pkg/k8s/labels.go

```diff
@@ -108,6 +108,10 @@ func GetControllerNs(pod *coreV1.Pod) string {
 	return pod.Labels[ControllerNSLabel]
 }
 
+func GetControllerVersion(pod *coreV1.Pod) string {
+	return pod.Annotations[ProxyVersionAnnotation]
+}
+
```

Comment:
And if you do revise the hint implementation as suggested, then this func will be unused and you can remove it.